### PR TITLE
ddl: move the code out of syncer

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: ref #4287 

### What is changed and how it works?

- move handleQueryEvent out of syncer

#### Tests 

 - Original test

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by GMHDBJD** [comment]
cc @lance6716

**Review by buchuitoudegou — APPROVED** [review]

File: dm/syncer/shardddl.go [hunk 1]
```diff
@@ -0,0 +1,849 @@
...
+
+	switch ddl.s.cfg.ShardMode {
+	case "":
+		return ddl.strategy.handleDDL(qec)
+	case config.ShardOptimistic:
+		return ddl.strategy.handleDDL(qec)
+	case config.ShardPessimistic:
+		return ddl.strategy.handleDDL(qec)
+	}
+	return errors.Errorf("unsupported shard-mode %s, should not happened", ddl.s.cfg.ShardMode)
```
Comment (okJiang) [inline dm/syncer/shardddl.go:316]: prefer

```go
if ddl.s.cfg.ShardMode not in ["", config.ShardOptimistic, config.ShardPessimistic]
	return errors.Errorf("unsupported shard-mode %s, should not happened", ddl.s.cfg.ShardMode)
return ddl.strategy.handleDDL(qec)
```

**Review by okJiang — COMMENTED** [review]
seems that your `refactorDDL` does not finish(exist some TODO)

Using feature branch develop is better?

Rest LGTM

**Review by okJiang — APPROVED** [review]

**Pull request merged**